airflow-core/src/airflow/api/common/utils.py

```diff
@@ -0,0 +1,26 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from fastapi import HTTPException, status
+
+if TYPE_CHECKING:
+    from airflow.models.dag import DAG
+
+
+def get_dag_from_dag_bag(dag_bag, dag_id: str) -> DAG:
```

Comment:
dag_bag can be injected from the request dependency, you don't need to pass it around.